src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -334,6 +330,34 @@ private async Task RunLoop(CancellationToken cancellationToken)
                 });
         }
 
+        private async Task RunGoal( AzureDevOpsTimelineOptions options)
+        { 
+            HttpClient client = new HttpClient();
+            string url = "https://raw.githubusercontent.com/dotnet/arcade/master/eng/GoalTimePerDefinition.json";
```

Comment:
How are you intending to represent that a single definition may be used for multiple release branches? ie, is the model you've implemented implies that corefx on the .NET Core 5 dev channel is supposed to have the same goal time as corefx on the .NET Core 3 channel?  I don't think that's what you intend.  Instead, there needs to be some way to specify a channel (or branch) in the json file that is associated with the definition, or the json file would need to be read from multiple branches of Arcade.  I think the former is more likely what we need.

Reply:
I tried both the scenarios 

1) Added channelId to the goal table. For this I connected Timelinebuilds(Perday in power BI report) through definition id as the primary key.  I couldn't connect to the channel table since TimelineBuild_perDay table is already connected to channel table. So this would not make sense. Also power bi will allow only one active relationship. 

2) I added sourcebranch Info to the goal table. Also I made sure to add different goals to each branch. It was easy to establish relationship between goal table and timeline build table. But I see a concern. 

In the power bi report now I see this 

![image](https://user-images.githubusercontent.com/47157394/67997103-c87f3680-fc0f-11e9-851e-783702642bcb.png)

In the above scenario I have picked .Net 3 Tools channel, but there are 2 different branches for this result. I can solve this by slicing through branch. That way I can keep the existing functionality in the report that represents if the goal is met or not. Let me know your thoughts.

Reply:
I had expected this file to get uploaded as part of the build, and we fetched it here.  Fetching the file from github directly is problematic.  The current state of this file might not have anything to do with the build (especially from non-master branches).

Are we going to put ALL the goals for every repository in this repository?  I assume this file was only going to describe arcade... not that arcade would have a reference to every file.  Tying arcade to everyones build seems like bad coupling.  Every time someone changes their build, they now need to come here and change arcade... which will flow to them and cause another build... but builds before that change will cause problems...

Reply:
As it is, there is little point to put this data in kusto... it's just going to repeat the same row information every half hour forever.